dotnet/src/Microsoft.AutoGen/Core.Grpc/GrpcAgentRuntime.cs

```diff
@@ -257,10 +260,22 @@ private async ValueTask HandlePublish(CloudEvent evt, CancellationToken cancella
                 var agent = await this._agentsContainer.EnsureAgentAsync(recipient);
 
                 // give the serializer a second chance to have been registered
-                serializer ??= SerializationRegistry.GetSerializer(typeName) ?? throw new Exception($"Could not find a serializer for message of type {typeName}");
-                message ??= serializer.Deserialize(evt.ProtoData);
+                serializer ??= SerializationRegistry.GetSerializer(typeName);
 
-                await agent.OnMessageAsync(message, messageContext);
+                if (serializer != null)
+                {
+                    message ??= serializer.Deserialize(evt.ProtoData);
+                    await agent.OnMessageAsync(message, messageContext);
+                }
+                else if (_strictMessageDeserialization)
```

Comment:
this feels like a workaround - I think we should get together soon and all discuss a desired goal state/spec for the medium term of how we want this to work